DDF-1202 Updating configuration deletion to properly clear, simplifying logic of new configurations

andrewkfiedler opened the pull request:
- Service.view.js:
  - Updated the "remove configuration" method to wait for the delete to happen, then to refresh the configurations and close.
  - Refactored the "new configuration" method to be much more straightforward.  The previous logic was convoluted.
- Service.js:
  - Updated the "destroy" method to not have a useless Deferred, reformatted the deleteUrl for readability, removed useless done and fail functions from the request.
  - Added the "initialize from model" method, so that you can pass in a model and have it determine how to initialize.
  - Updated the other "initialize from" type methods so that they would return a reference to "this", so method chaining is possible.
  - Various changes to utilize method chaining / optimize.

<!-- Reviewable:start -->

[<img src="https://reviewable.io/review_button.png" height=40 alt="Review on Reviewable"/>](https://reviewable.io/reviews/codice/ddf-admin/79)

<!-- Reviewable:end -->

andrewkfiedler committed:
DDF-1202 Updating configuration deletion to properly clear, simplifying logic of new configurations

 - Service.view.js:
  - Updated the "remove configuration" method to wait for the delete to happen, then to refresh the configurations and close.
  - Refactored the "new configuration" method to be much more straightforward.  The previous logic was quite convoluted.
 - Service.js:
  - Updated the "destroy" method to not have a useless Deferred, reformatted the deleteUrl for readability, removed useless done and fail functions from the request.
  - Added the "initialize from model" method, so that you can pass in a model and have it determine how to initialize.
  - Updated the other "initialize from" type methods so that they would return a reference to "this", so method chaining is possible.
  - Various changes to utilize method chaining / optimize.

andrew-fiedler commented:
@stustison @shaundmorris

shaundmorris commented:
Review status: :shipit: all files reviewed, all discussions resolved, all commit checks successful.
Reviewed files:
- ui/src/main/webapp/js/models/Service.js @ r1
- ui/src/main/webapp/js/views/configuration/Service.view.js @ r1

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-admin/79)
<!-- Sent from Reviewable.io -->

shaundmorris commented:
I verified the build was successful and was able to successfully delete configurations.

---

Review status: :shipit: all files reviewed, all discussions resolved, all commit checks successful.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf-admin/79)
<!-- Sent from Reviewable.io -->

pklinef commented on ui/src/main/webapp/js/models/Service.js:
```diff
@@ -133,35 +133,42 @@ define(['backbone', 'jquery','backboneassociations'],function (Backbone, $) {
-            this.set({"name":msf.get("name")});
-            this.get('properties').set({"service.factoryPid": msf.get("id")});
-            this.initializeFromService(msf);
+            this.set({"fpid": msf.get("id")})
```
Instead of calling set twice, call it once and pass in an object with both properties in it.  This will reduce the event change on model changes.

pklinef commented:
:+1:

andrewkfiedler committed:
DDF-1202 Taking advantage of passing objects to 'set' calls

 - Updated initializeFromMSF and intiailizeFromMetatype to take advantage of being able to pass objects into 'set' calls.

shaundmorris commented:
Re-verified build and testing in AdminUI
